packages/nextcloud/lib/src/helpers/spreed.dart

```diff
@@ -0,0 +1,199 @@
+import 'package:nextcloud/src/api/core.openapi.dart' as core;
+import 'package:nextcloud/src/api/spreed.openapi.dart' as spreed;
+import 'package:version/version.dart';
+
+/// Version of the spreed app supported
+const supportedVersion = 17;
+
+/// Extension for checking if spreed is supported
+extension SpreedVersionSupported on spreed.Client {
+  /// Check if the spreed app version is supported by this client
+  ///
+  /// Also returns the supported version number
+  Future<(bool, int)> isSupported(final core.OcsGetCapabilitiesResponseApplicationJson_Ocs_Data capabilities) async {
+    final version = capabilities.capabilities.spreedPublicCapabilities?.spreed.version;
+    return (
+      version != null && Version.parse(version).major == supportedVersion,
+      supportedVersion,
+    );
+  }
+}
+
+/// Conversation types.
+///
+/// Use [value] to get the integer representation that is used in the API.
+/// See https://github.com/nextcloud/spreed/blob/master/lib/Room.php.
+enum RoomType {
+  /// "One to one".
+  oneToOne,
+
+  /// Group.
+  group,
+
+  /// Public.
+  public,
+
+  /// Changelog.
+  changelog,
+
+  /// Former "One to one" (When a user is deleted from the server or removed from all their conversations, `1` "One to one" rooms are converted to this type).
+  oneToOneFormer,
+
+  /// Note to self.
+  noteToSelf;
+
+  /// Integer representation of the [ParticipantType].
+  int get value => index + 1;
+}
+
+/// Types of chat messages.
+///
+/// Use [name] to get the string representation that is used in the API.
```

Comment:
Not implemented that way.

Reply:
No this is right? See the unit tests where this is used.

Reply:
My bad. This time I forgot about the `name` getters on enums :sweat_smile: